src/test/ui/typeck/issue-98982.stderr

```diff
@@ -0,0 +1,30 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-98982.rs:2:5
+   |
+LL |   fn foo() -> i32 {
+   |               --- expected `i32` because of return type
+LL | /     for i in 0..0 {
+LL | |
+LL | |         return i;
+LL | |     }
+   | |_____^ expected `i32`, found `()`
+
+note: the function expects a value to always be returned, but loops might run zero times
+  --> $DIR/issue-98982.rs:2:5
+   |
+LL |       for i in 0..0 {
+   |       ^------------
+   |       |
+   |  _____this might have zero elements to iterate on
+   | |
+LL | |
+LL | |         return i;
+   | |         -------- if the loop doesn't execute, this value would never get returned
+LL | |     }
+   | |_____^
```

Comment:
It would be better to skip the span that points at the whole `for` loop, so this is like this instead:

```
note: the function expects a value to always be returned, but loops might run zero times
  --> $DIR/issue-98982.rs:2:5
   |
LL |     for i in 0..0 {
   |     ^^^^^^^^^^^^^ this might have zero elements to iterate on
   |
LL |
LL |         return i;
   |         -------- if the loop doesn't execute, this value would never get returned

```